### PR TITLE
chore: add max lines linting rules and refactor

### PR DESCRIPTION
**Pull request by jaredcwhite**
## Issue Overview

This PR addresses #2337 

- [x] This change addresses the issue in full
- [ ] This change addresses only certain aspects of the issue
- [ ] This change is a dependency for another issue
- [ ] This change has a dependency from another issue

## Description

Configures eslint with two new rules: max-lines-per-function and max-lines (per file). The intention is to encourage breaking up large single functions (sometimes with nested functions) into smaller discrete functions and perhaps additional files, so code blocks are simpler to maintain and easier to unit test—and possibly improve performance or memory usage as well.

FYI, the numbers I chose for the rules was after tweaking the values a few different times to find a sweet spot between not enough stuff/too much stuff getting caught in our existing codebase. Also the way React Hook Form-related components get written ends up with really long functions much of the time, so I'm not sure how much we can mitigate against that, but it does seems like there's some room for improvement in a few places.

## How Can This Be Tested/Reviewed?

Obviously we can tell if the linter & tests all run as expected, but in the case of refactored files we should try to QA the pages or API endpoints which were modified.

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have reviewed the changes in a desktop view
- [ ] I have reviewed the changes in a mobile view
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [x] I have assigned reviewers
- [ ] I have run `yarn generate:client` and/or created a migration if I made backend changes that require them
- [ ] I have exported any new pieces added to ui-components
- [x] My commit message(s) is/are polished, and any breaking changes are indicated in the message and are well-described
- [ ] Commits made across packages purposefully have the same commit message/version change, else are separated into different commits

## Reviewer Notes:

Steps to review a PR:

- Read and understand the issue
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Also review the acceptance criteria on the Netlify deploy preview (noting that these do not yet include any backend changes made in the PR)
- Either explicitly ask a clarifying question, request changes, or approve the PR if there are small remaining changes but the PR is otherwise good to go

## On Merge:

If you have one commit and message, squash. If you need each message to be applied, rebase and merge.

### CODE REVIEW

**Comment by emilyjablonski** [comment]
Is this ready for review?

File: backend/core/src/shared/hmi-data.ts [hunk 1]
```diff
@@ -0,0 +1,220 @@
+import { Unit } from "../units/entities/unit.entity"
```
Comment (seanmalbert) [inline backend/core/src/shared/hmi-data.ts:1]: @jaredcwhite , just want to confirm, that nothing was changed here, except for pulling these pieces out of unit-transformations to get the file size down under 400 lines, correct?

Comment (seanmalbert) [reply backend/core/src/shared/hmi-data.ts:1]: Commented too soon, looks like `amiPercentagesAcrossUnits`, `findAmiValueInChart`, and `setupHmiHeaders` was pulled out

**Comment by jaredcwhite** [comment]
@emilyjablonski yes, sorry I forgot to add the needs review label.

**Comment by YazeedLoonat** [comment]
@jaredcwhite could you explain how you got to your cut offs for these?

**Comment by YazeedLoonat** [comment]
Hey @jaredcwhite @seanmalbert @emilyjablonski 
I feel like that limits for max lines per function and max lines per file are somewhat arbitrary in this pr. I tried to look into if there are standards set out by some of the larger open source projects I'm familiar with and none of them seem to use these rules at all. 

https://github.com/mui-org/material-ui/blob/master/.eslintrc.js
https://github.com/microsoft/TypeScript/blob/main/.eslintrc.json
https://github.com/facebook/react/blob/main/.eslintrc.js


I'm not entirely sure I agree with setting these rules in the first place since it seems like in a lot of places already we had to set up files and functions to disable/ignore the rules anyway 

I do think it is a good idea to have separated out the functions and files that you already did in this pr. I feel like code reviews should catch if a function can be broken up, or a file can be split up and we won't have to worry about a seemingly arbitrary rule guiding that

**Comment by emilyjablonski** [comment]
I'm in agreement with Yazeed here - I think it's an awesome best practice that we can and should incorporate in PR reviews but I don't see a ton of value out of enforcing it this way, considering there are places we want to disable it and it kind of depends on the context.

**Comment by emilyjablonski** [comment]
Seems like the PR review resulted in passing on this, and given it's been almost a year going to close, lmk if you'd prefer a different resolution!

**Pull request closed without merging**